src/pages/RegisterPage/RegisterPage.js

```diff
@@ -0,0 +1,100 @@
+import TextInput from "../../components/TextInput/TextInput";
+import classes from "./RegisterPage.module.scss";
+
+const schema = yup
+  .object()
+  .shape({
+    name: yup.string().required("name is required").required(),
+    email: yup.string().email("invalid email").required(),
+    password: yup.string().required("recipe must have a name"),
+    password2: yup.string().required("repeat password")
+  })
+  .required();
+
+const RegisterPage = () => {
+  const {
+    register,
+    handleSubmit,
+    formState: { errors }
+  } = useForm({ resolver: yupResolver(schema) });
+
+  const navigate = useNavigate();
+
+  const onSubmit = (data) => {
+    fetch("/api/dishes", { method: "POST", body: data });
+  };
+
+  return (
+    <div className={classes.pageWrapper}>
+      <SubpageTitle subpageTitle="Register" />;
+      <div className={classes.formWrapper}>
+        <form
+          style={{ display: "flex", flexDirection: "column", gap: "2rem" }}
+          onSubmit={handleSubmit(onSubmit)}
+        >
+          <div className={classes.formRowItems}>
+            <Label labelName="name" />
+            <TextInput
+              placeholder="name"
+              {...register("name", { required: true })}
+            />
+            <span style={{ gridColumn: "span 2" }}>
+              {errors.email?.type === "required" && "name is required"}
+            </span>
+          </div>
+          <div className={classes.formRowItems}>
+            <Label labelName="email" />
+            <TextInput
+              placeholder="email"
+              {...register("email", { required: true })}
+            />
+            <span style={{ gridColumn: "span 2" }}>
+              {errors.email?.type === "required" && "invalid email format"}
+            </span>
+          </div>
+          <div className={classes.formRowItems}>
+            <Label labelName="pass" />
+            <TextInput
+              placeholder="password"
+              {...register("password", { required: true })}
+            />
+            <span style={{ gridColumn: "span 2" }}>
+              {errors.password?.type === "required" && "password is required"}
+            </span>
+          </div>
+          <div className={classes.formRowItems}>
+            <Label labelName="pass" />
+            <TextInput
+              placeholder="confirm password"
+              {...register("password2", { required: true })}
+            />
+            <span style={{ gridColumn: "span 2" }}>
+              {errors.password?.type === "required" &&
+                "please confirm password"}
+            </span>
+          </div>
+          <Button
+            submit
+            text="Register"
+            onClick={navigate(`/my_cookbook/login`)}
+          />
+        </form>
+        <div className={classes.formSubmitItems}>
+          <h1>Already registered?</h1>
+        </div>
+        <Button text="Login" onClick={navigate(`/my_cookbook/login`)} />
```

Comment:
Usually the `Already registered?` string is also a link to the `/register` page. Having two identical buttons can be bad for UX.
You also used h1 for this text. It's not a heading at all and it for sure shouldn't be the most important heading on the page.